runtime/jcl/common/mgmtthread.c

```diff
@@ -1210,32 +1184,28 @@ initIDCache(JNIEnv *env)
 
 	mid = (*env)->GetMethodID(env, gcls, "<init>", "(Ljava/lang/Object;)V");
 	if (!mid) {
-		err = JNI_ERR;
 		goto initIDCache_fail;
 	}
 	JCL_CACHE_SET(env, MID_openj9_management_internal_LockInfoBase_init, mid);
 
 	cls = (*env)->FindClass(env, "java/lang/StackTraceElement");
 	if (!cls) {
-		err = JNI_ERR;
 		goto initIDCache_fail;
 	}
 	if (!(gcls = (*env)->NewGlobalRef(env, cls))) {
 		err = JNI_ENOMEM;
 		goto initIDCache_fail;
 	}
-	(*env)->DeleteLocalRef(env, cls);
 	JCL_CACHE_SET(env, CLS_java_lang_StackTraceElement, gcls);
 
 	mid = (*env)->GetMethodID(env, gcls, "isNativeMethod", "()Z");
 	if (!mid) {
-		err = JNI_ERR;
 		goto initIDCache_fail;
 	}
 	JCL_CACHE_SET(env, MID_java_lang_StackTraceElement_isNativeMethod, mid);
 
 	(*env)->DeleteLocalRef(env, oom);
```

Comment:
@gacholio Should this call to `DeleteLocalRef()` stay or go? It doesn't seem too helpful given we're about to return.

Reply:
I'd prefer to see it go. We should technically delete global refs if the overall init fails.

Reply:
done, removed it.